### PR TITLE
fix(integer): support integer field type

### PR DESCRIPTION
**Pull request by jguddas**
SON Schemas fields with `type: "integer"` allow values like `1.5`. 
Now in this PR they are considered invalid. If you want `1.5` to be allowed, use `type: "number"`. 

Example:

```js
{
  properties: {
    tabs: {
      title: 'Tabs',
      description: 'How many open tabs do you have?',
      'x-jsf-presentation': {
        inputType: 'number', // the inputType can remain number
      },
      minimum: 1,
      type: 'integer', // New!
    },
  },
};
```

Before:
- `{tabs: 2.5}` // valid
- `{tabs: 2.5}` // error:  "Must not contain decimal points. E.g. 5 instead of 5.5"

### CODE REVIEW

File: src/tests/helpers.js [hunk 1]
```diff
@@ -919,6 +930,13 @@ export const schemaInputTypeSelectMultipleOptional = JSONSchemaBuilder()
   })
   .build();
 
+export const schemaInputTypeInteger = JSONSchemaBuilder()
```
Comment (sandrina-p) [inline src/tests/helpers.js:933]: Some of the comments I mentioned at #10 are applied here too. I'll wait for you to fix those, re-applied them here, and then we can merge this one. It looks great!

**Comment by sandrina-p** [comment]
Hi @jguddas! Could you update this with the main, and then we can merge it?

**Comment by jguddas** [comment]
> Hi @jguddas! Could you update this with the main, and then we can merge it?

Done.

**Comment by sandrina-p** [comment]
Hi @jguddas, forgive us the radio silence here. I'm picking this again today. I'm updating your branch with main, yet I noticed a test failing. I'll fix it tomorrow, and release a dev version to run against our internal CI/CD, and merge it :)

Once again, thanks for your contribution here!

File: src/tests/createHeadlessForm.test.js [hunk 2]
```diff
@@ -994,6 +995,41 @@ describe('createHeadlessForm', () => {
       expect(fieldOptions).toEqual([]);
     });
 
+    it('support "integer" field type', () => {
```
Comment (sandrina-p) [inline src/tests/createHeadlessForm.test.js:998]: Hi @jguddas, forgive us the radio silence here. I'm picking this again today. I'm updating your branch with main, yet I noticed a test failing. I'll fix it tomorrow, and release a dev version to run against our internal CI/CD, and merge it :)

Once again, thanks for your contribution here!

Comment (sandrina-p) [reply src/tests/createHeadlessForm.test.js:998]: For reference: Internal [PR running the CI/CD](https://gitlab.com/remote-com/employ-starbase/dragon/-/merge_requests/33698)

Comment (sandrina-p) [reply src/tests/createHeadlessForm.test.js:998]: Internal CI is green, merging this 🎉

**Comment by brennj** [comment]
Approving!

**Comment by sandrina-p** [comment]
Merged and released `@remoteoss/json-schema-form@0.11.8-beta.0`. Once again, thank you @jguddas 🎉

**Pull request merged**